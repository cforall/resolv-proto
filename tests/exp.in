// Based on exponential failure case described on p.7 of Aaron's Comp II

ctor-v #voidptr
dtor-v #voidptr

ctor-b #box<T> | ctor T | dtor T
dtor-b #box<T> | ctor T | dtor T

f-v #voidptr
f-b #box<S> | ctor S | dtor S

#voidptr conv-v #voidptr
Any conv-b #voidptr

Q make | ctor Q | dtor Q

%%

conv( #voidptr )
f( make() )
f( conv( #voidptr ) )
