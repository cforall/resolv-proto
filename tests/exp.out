[#voidptr / (0,0,0,0,0)] conv-v( #voidptr )
[Void / (0,1,1,-2,0)] f-v( make( ) )
[Void / (0,0,0,0,0)] f-v( conv-v( #voidptr ) )
